pkg/registry/extensions/daemonset/strategy.go

```diff
@@ -87,12 +87,12 @@ func (daemonSetStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, ol
 	//
 	// TODO: Any changes to a part of the object that represents desired state (labels,
 	// annotations etc) should also increment the generation.
-	if !reflect.DeepEqual(oldDaemonSet.Spec.Template, newDaemonSet.Spec.Template) {
+	if !apiequality.Semantic.DeepEqual(oldDaemonSet.Spec.Template, newDaemonSet.Spec.Template) {
 		newDaemonSet.Spec.TemplateGeneration = oldDaemonSet.Spec.TemplateGeneration + 1
 		newDaemonSet.Generation = oldDaemonSet.Generation + 1
 		return
 	}
-	if !reflect.DeepEqual(oldDaemonSet.Spec, newDaemonSet.Spec) {
+	if !apiequality.Semantic.DeepEqual(oldDaemonSet.Spec, newDaemonSet.Spec) {
```

Comment:
Open a separate issue to replace reflect.DeepEqual in all of the other registries